commit 702a7c7609bec3a940b6a46b0d6ab9ce45274580
Merge: 053fe57ac249 b9889ed1ddec
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Sat Dec 12 11:34:10 2009 -0800

    Merge branch 'sched-fixes-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/tip/linux-2.6-tip
    
    * 'sched-fixes-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/tip/linux-2.6-tip: (21 commits)
      sched: Remove forced2_migrations stats
      sched: Fix memory leak in two error corner cases
      sched: Fix build warning in get_update_sysctl_factor()
      sched: Update normalized values on user updates via proc
      sched: Make tunable scaling style configurable
      sched: Fix missing sched tunable recalculation on cpu add/remove
      sched: Fix task priority bug
      sched: cgroup: Implement different treatment for idle shares
      sched: Remove unnecessary RCU exclusion
      sched: Discard some old bits
      sched: Clean up check_preempt_wakeup()
      sched: Move update_curr() in check_preempt_wakeup() to avoid redundant call
      sched: Sanitize fork() handling
      sched: Clean up ttwu() rq locking
      sched: Remove rq->clock coupling from set_task_cpu()
      sched: Consolidate select_task_rq() callers
      sched: Remove sysctl.sched_features
      sched: Protect sched_rr_get_param() access to task->sched_class
      sched: Protect task->cpus_allowed access in sched_getaffinity()
      sched: Fix balance vs hotplug race
      ...
    
    Fixed up conflicts in kernel/sysctl.c (due to sysctl cleanup)

diff --cc kernel/sysctl.c
index 9327a26765c5,b9e5a45f1e28..554ac4894f0f
--- a/kernel/sysctl.c
+++ b/kernel/sysctl.c
@@@ -260,7 -273,8 +264,7 @@@ static struct ctl_table kern_table[] = 
  		.data		= &sysctl_sched_min_granularity,
  		.maxlen		= sizeof(unsigned int),
  		.mode		= 0644,
- 		.proc_handler	= sched_nr_latency_handler,
 -		.proc_handler	= &sched_proc_update_handler,
 -		.strategy	= &sysctl_intvec,
++		.proc_handler	= sched_proc_update_handler,
  		.extra1		= &min_sched_granularity_ns,
  		.extra2		= &max_sched_granularity_ns,
  	},
@@@ -269,7 -284,8 +273,7 @@@
  		.data		= &sysctl_sched_latency,
  		.maxlen		= sizeof(unsigned int),
  		.mode		= 0644,
- 		.proc_handler	= sched_nr_latency_handler,
 -		.proc_handler	= &sched_proc_update_handler,
 -		.strategy	= &sysctl_intvec,
++		.proc_handler	= sched_proc_update_handler,
  		.extra1		= &min_sched_granularity_ns,
  		.extra2		= &max_sched_granularity_ns,
  	},
@@@ -278,7 -295,8 +282,7 @@@
  		.data		= &sysctl_sched_wakeup_granularity,
  		.maxlen		= sizeof(unsigned int),
  		.mode		= 0644,
- 		.proc_handler	= proc_dointvec_minmax,
 -		.proc_handler	= &sched_proc_update_handler,
 -		.strategy	= &sysctl_intvec,
++		.proc_handler	= sched_proc_update_handler,
  		.extra1		= &min_wakeup_granularity_ns,
  		.extra2		= &max_wakeup_granularity_ns,
  	},
@@@ -287,24 -306,34 +291,28 @@@
  		.data		= &sysctl_sched_shares_ratelimit,
  		.maxlen		= sizeof(unsigned int),
  		.mode		= 0644,
- 		.proc_handler	= proc_dointvec,
 -		.proc_handler	= &sched_proc_update_handler,
++		.proc_handler	= sched_proc_update_handler,
+ 		.extra1		= &min_sched_shares_ratelimit,
+ 		.extra2		= &max_sched_shares_ratelimit,
+ 	},
+ 	{
 -		.ctl_name	= CTL_UNNUMBERED,
+ 		.procname	= "sched_tunable_scaling",
+ 		.data		= &sysctl_sched_tunable_scaling,
+ 		.maxlen		= sizeof(enum sched_tunable_scaling),
+ 		.mode		= 0644,
 -		.proc_handler	= &sched_proc_update_handler,
 -		.strategy	= &sysctl_intvec,
++		.proc_handler	= sched_proc_update_handler,
+ 		.extra1		= &min_sched_tunable_scaling,
+ 		.extra2		= &max_sched_tunable_scaling,
  	},
 -
  	{
 -		.ctl_name	= CTL_UNNUMBERED,
  		.procname	= "sched_shares_thresh",
  		.data		= &sysctl_sched_shares_thresh,
  		.maxlen		= sizeof(unsigned int),
  		.mode		= 0644,
 -		.proc_handler	= &proc_dointvec_minmax,
 -		.strategy	= &sysctl_intvec,
 +		.proc_handler	= proc_dointvec_minmax,
  		.extra1		= &zero,
  	},
- 	{
- 		.procname	= "sched_features",
- 		.data		= &sysctl_sched_features,
- 		.maxlen		= sizeof(unsigned int),
- 		.mode		= 0644,
- 		.proc_handler	= proc_dointvec,
- 	},
  	{
 -		.ctl_name	= CTL_UNNUMBERED,
  		.procname	= "sched_migration_cost",
  		.data		= &sysctl_sched_migration_cost,
  		.maxlen		= sizeof(unsigned int),